services/src/test/java/org/exoplatform/news/NewsTargetingImplTest.java

```diff
@@ -111,5 +116,82 @@ public void shouldReturnReferencedTargets() {
     assertEquals("sliderNews", newsTargetingEntities.get(0).getName());
   }
 
+  @Test
+  public void shouldReturnSelectedTargets() {
+    // Given
+    NewsTargetingServiceImpl newsTargetingService = new NewsTargetingServiceImpl(metadataService);
+    Metadata sliderNews = new Metadata();
+    sliderNews.setName("sliderNews");
+    sliderNews.setCreatedDate(100);
+    HashMap<String, String> sliderNewsProperties = new HashMap<>();
+    sliderNewsProperties.put("referenced", "true");
+    sliderNews.setProperties(sliderNewsProperties);
+    sliderNews.setId(1);
+
+    List<MetadataItem> metadataItems = new LinkedList<>();
+    MetadataItem metadataItem = new MetadataItem();
+    metadataItem.setCreatedDate(100);
+    metadataItem.setCreatorId(1);
+    metadataItem.setId(1);
+    metadataItem.setObjectId("123456");
+    metadataItem.setMetadata(sliderNews);
+    metadataItems.add(metadataItem);
+
+    NewsTargetObject newsTargetObject = new NewsTargetObject("news", "123456", null);
+
+    // When
+    when(metadataService.getMetadataItemsByMetadataTypeAndObject(NewsTargetingService.METADATA_TYPE.getName(), newsTargetObject)).thenReturn(metadataItems);
+    List<String> selectedTargets = newsTargetingService.getTargetsByNewsId("123456");
+
+    // Then
+    assertNotNull(selectedTargets);
+    assertEquals(1, selectedTargets.size());
+    assertEquals("sliderNews", selectedTargets.get(0));
+  }
+
+  @Test
+  public void testLinkNewstoTargets() throws Exception {
```

Comment:
```suggestion
  public void testSetNewsTargets() throws Exception {
```
Not sure that this is a test for setNewsTargets